### PR TITLE
generate and report telemetry

### PR DESCRIPTION
**Pull request by Rader**

### CODE REVIEW

File: component/telemetry.go [hunk 1]
```diff
@@ -47,3 +54,81 @@ func (tc *TelemetryComponent) SaveUsageData(ctx context.Context, usage telemetry
...
+	usage.UUID = uuid.String()
+	usage.RecordedAt = time.Now()
+	usage.Version = ""
+	usage.InstallationType = ""
+	usage.ActiveUserCount, err = tc.getUserCnt(ctx)
+	if err != nil {
+		return usage, fmt.Errorf("failed to get user count: %w", err)
+	}
+	usage.Edition = ""
+	usage.HistoricalMaxUsers = 0
```
Comment (ganisback) [inline component/telemetry.go:74]: 无用的字段？

File: common/config/config.go [hunk 2]
```diff
@@ -150,6 +150,11 @@ type Config struct {
 	MultiSync struct {
 		// Enabled bool `envconfig:"STARHUB_SERVER_MULTI_SYNC_ENABLED" default:"false"`
 	}
+
+	Telemetry struct {
+		Enable    bool   `envconfig:"STARHUB_SERVER_TELEMETRY_ENABLE" default:"true"`
+		ReportURL string `envconfig:"STARHUB_SERVER_TELEMETRY_URL" default:"http://hub.opencsg.com/api/v1/telemetry"`
```
Comment (ganisback) [inline common/config/config.go:156]: API was not implemented

**Comment by ganisback** [comment]
How can we view the local telemetry? only view by database?

File: common/config/config.go (hunk 2 above)
Comment (Rader) [reply common/config/config.go:156]: see this PR: https://github.com/OpenCSGs/csghub-server/pull/78

File: component/telemetry.go (hunk 1 above)
Comment (Rader) [reply component/telemetry.go:74]: 基本字段是参考 gitlab 上报的数据，有些目前没值，后续会实现。

**Review by SeanHH86 — APPROVED** [review]
LGTM

**Pull request merged**